applications/nrf_desktop/doc/dvfs.rst

```diff
@@ -0,0 +1,116 @@
+.. _nrf_desktop_dvfs:
+
+DVFS module
+###########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The nRF Desktop Dynamic Voltage and Frequency Scaling (DVFS) module is responsible for switching the frequency and voltage according to the application needs.
+The module chooses the highest frequency required by the application states and selects voltage accordingly.
+DVFS is critical for balancing performance and power consumption, especially in battery-operated devices.
+
+In the nRF Desktop, the DVFS module enables the system to use the lowest possible frequency and voltage when the application performs less computation-challenging activities such as Bluetooth LE advertising.
+However, if performing more computation-consuming tasks nRF Desktop DVFS module will dynamically change to a higher frequency to increase performance.
+
+The nRF Desktop DVFS module requests clock frequency change using API of the clock control driver.
+The driver is responsible for choosing the optimal clock frequency and voltage.
```

Comment:
The driver actually seems not to know anything about the voltage. It's more like: the driver controls frequency and sysctrl core aligns voltage when setting the frequency.

Reply:
That's no entirely true but I don't have time to argue. Let it be your way, if you must

Reply:
I am not sure about responsibility split between nrfs service on app core and code running on sysctrl core (so I might be inaccurate here). Still, if we inform what happens under the hood, we should not provide false impression (the clock control driver really does not choose voltage :) ).